datafusion/core/tests/physical_optimizer/limited_distinct_aggregation.rs

```diff
@@ -0,0 +1,443 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Tests for the limited distinct aggregation optimizer rule
+
+use std::sync::Arc;
```

Comment:
I think if possible it would be best to keep these tests  alongside the code in `limited_distinct_aggregation.rs` though I know in some cases the tests have too  many dependencies to the core crate for that to be feasible

Even though most of the the use of the core crate is `SessionContext`  which could be removed, I think the use of ParquetExec in the tests makes moving these tests the best option at this point

Reply:
I agree with this

Reply:
Nice approach! We could try to eliminate `SessionContext` in the test later.

Reply:
> Nice idea! We could try to eliminate SessionContext in the test later.

Thanks -- I think we could remove `SessionContext` -- the problem is that `ParquetExec` is also in the core datafusion crate which is not nearly as easy to remove from the tests I don't think. 

We could make a `StubParquetExec` or something for testing the physical optimizer passes (which would allow us to keep the tests in the datafusion-physical-optimizer), but then I worry that the real parquet exec would diverge from the stub and the tests would not catch regressions.